integration_test/scripts/build-adapter-image.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+# imageProject commitShortSha
+
+export PATH=$PATH:/opt/gradle/gradle-6.7/bin
+base_name="gcr.io/${1}/healthcare-api-dicom-dicomweb-adapter-"
+
+build_adapter () {
+  gradle dockerBuildImage
+  adapter_name=${base_name}${1}
+  version=$(gradle printVersion  | grep -Eo "[0-9]{1,3}\.[0-9]{1,3}" | head -1)
+  postfix=$(gcloud container images list-tags "${adapter_name}" --sort-by=~timestamp | grep " $version" |  grep -Eo "([0-9]{1,3}\.){2}[0-9]{1,3}" | head -1 | grep -Eo "[0-9]{1,3}" | tail -1)
+  postfix=$(($postfix+1))
+  gradle dockerBuildImage -Pdocker_tag=${adapter_name}:$version.$postfix
+  gradle dockerBuildImage -Pdocker_tag=${adapter_name}:$2
+}
+apt-get -qq update
+apt-get install -y gnupg2
+
+gradle build &&
+cd ./import &&
+build_adapter import $2 &&
+cd ../export &&
+build_adapter export $2
```

Comment:
Why do we need "&&" here but not above with the apt-get statements?

Reply:
we need && so that the commands are executed in a chain, for example, if an error is returned when adapter builds, then the test step will end with an error

Reply:
Right, but it does not seem to be consistent, just wondering why?

apt-get -qq update          <---- no "&&" after here
apt-get install -y gnupg2 <---- or here

gradle build &&
cd ./import &&
build_adapter import $2 &&
cd ../export &&
build_adapter export $2

Reply:
Execution of the command **apt-get -qq update** is not critical for building the adapter, even if during its execution there are errors, they are not always related to the commands that we need.
Execution of **apt-get install -y gnupg2** is mandatory and if the execution of this command gives a result not 0, then we get an error when executing the commands below. But we can add && after **apt-get install -y gnupg2** to ensure that gnupg2 is installed

Reply:
I think I get it. So it's only the last command that we check the error code of.

Reply:
Can we use 
set -e
and 
set -o pipefail

at the start of the bash script to avoid the need for && everywhere? 

https://stackoverflow.com/questions/821396/aborting-a-shell-script-if-any-command-returns-a-non-zero-value

Reply:
Yes, thanks, it looks good and it works. If the some command ends with an error the entire test step fails

Reply:
Cool, thanks